core/src/main/java/org/apache/iceberg/PartitionsTable.java

```diff
@@ -45,9 +48,17 @@ public class PartitionsTable extends BaseMetadataTable {
     this.schema =
         new Schema(
             Types.NestedField.required(1, "partition", Partitioning.partitionType(table)),
-            Types.NestedField.required(2, "record_count", Types.LongType.get()),
-            Types.NestedField.required(3, "file_count", Types.IntegerType.get()),
-            Types.NestedField.required(4, "spec_id", Types.IntegerType.get()));
+            Types.NestedField.required(4, "spec_id", Types.IntegerType.get()),
+            Types.NestedField.required(
+                2,
+                "record_count",
+                Types.LongType.get(),
+                "data record count without applying the deletes"),
```

Comment:
Can we do a positive description like:  "count of records in data files".  Not sure "without applying deletes" sounds very clear.

If we add comments to "pos_delete_record_count" and "eq_delete_record_count". "count of records in position delete files", etc, I think that should be better?